src/CurrencyRateController.ts

```diff
@@ -85,31 +84,31 @@ export class CurrencyRateController extends BaseController<CurrencyRateConfig, C
 	 * @param currency - ISO 4217 currency code
 	 */
 	set currency(currency: string) {
-		this.activeCurrency = currency;
+		this.update({ currentCurrency: currency });
```

Comment:
Since `currency` is now maintained as state, it doesn't need a getter at all. This whole method can go away now.